src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDao.java

```diff
@@ -86,13 +88,19 @@ public MandateSearchDao(Jdbi jdbi) {
         queryMap.put("gatewayAccountExternalId", params.gatewayAccountExternalId.getValue());
         params.getCaseInsensitivePartialSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey(), "%" + e.getValue() + "%"));
         params.getCaseInsensitivePartialSearchTermsOnPayer().forEach(e -> queryMap.put(e.getKey(), "%" + e.getValue() + "%"));
+        params.getExactSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey(), e.getValue()));
+        params.getGreaterThanSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey() + "_from", e.getValue()));
+        params.getLessThanSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey() + "_to", e.getValue()));
         return queryMap;
     }
 
     private String generateQuery(Params params) {
         StringBuilder sb = new StringBuilder();
         params.getCaseInsensitivePartialSearchTermsOnMandate().forEach(e -> sb.append(format(" AND m.%s ILIKE :%s", e.getKey(), e.getKey())));
         params.getCaseInsensitivePartialSearchTermsOnPayer().forEach(e -> sb.append(format(" AND p.%s ILIKE :%s", e.getKey(), e.getKey())));
+        params.getExactSearchTermsOnMandate().forEach(e -> sb.append(format(" AND m.%s = :%s", e.getKey(), e.getKey())));
+        params.getGreaterThanSearchTermsOnMandate().forEach(e -> sb.append(format(" AND m.%s > :%s", e.getKey(), e.getKey() + "_from")));
```

Comment:
It took me more time than I care to admit to determine that this doesn’t introduce a risk of SQL injection (because `e.getKey()` comes from predefined constant strings defined by us rather than any remote input. I wish this was more obvious.